cmake/vif.cmake

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2022 The Chromium OS Authors
+# SPDX-License-Identifier: Apache-2.0
+
+# Generates USB-C VIF policies in XML format from device tree.
+set(GEN_VIF_SCRIPT ${ZEPHYR_BASE}/scripts/generate_usb_vif/generate_vif.py)
+set(DTS_COMPATIBLE usb-c-connector)
+set(VIF_XML ${PROJECT_BINARY_DIR}/vif.xml)
+set(CMD_GEN_VIF ${PYTHON_EXECUTABLE} ${GEN_VIF_SCRIPT}
+        --edt-pickle ${EDT_PICKLE}
+        --compatible ${DTS_COMPATIBLE}
+        --vif-out ${VIF_XML}
+        --board ${BOARD}
+        )
+
+message(STATUS "Generating vif.xml at ${VIF_XML}")
+
+add_custom_command(
+        OUTPUT ${VIF_XML}
+        DEPENDS ${EDT_PICKLE}
+        COMMAND ${CMD_GEN_VIF}
```

Comment:
not correct.

When used as post build command, it is true the COMMENT is not printed, ref: https://gitlab.kitware.com/cmake/cmake/-/issues/15461

but on a custom command like this, invoked through a custom target, then the `COMMENT` is printed.
Also note, splitting this into two commands are not safe because when ninja spawns multiple jobs, then the message and the command itself may be split because of other commands being executed between those two commands that you are now defining.

Hence your proposed solution is not safe.

Please use `COMMENT`.

Reply:
I tried putting COMMENT before COMMAND, but I didn't find the comment getting printed though the script gets executed, hence added the command with echo earlier. I concatenated the echo command with gen_vif command in latest revision.

Reply:
it's printed.

If you apply the following patch, it's there:
```
$ git diff
diff --git a/cmake/vif.cmake b/cmake/vif.cmake
index 94626fa134..917b3d1fb7 100644
--- a/cmake/vif.cmake
+++ b/cmake/vif.cmake
@@ -15,7 +15,8 @@ set(cmd_gen_vif ${PYTHON_EXECUTABLE} ${gen_vif_script}
 add_custom_command(
         OUTPUT ${vif_xml}
         DEPENDS ${EDT_PICKLE}
-        COMMAND ${cmd_gen_vif} && echo "-- Generated vif.xml at ${vif_xml}"
+        COMMAND ${cmd_gen_vif}
+        COMMENT "Generating vif.xml at ${vif_xml}"
 )
 
 add_custom_target(gen_vif ALL DEPENDS ${vif_xml})
$ ninja -Cbuild |grep vif
[4/178] Generating vif.xml at /projects/github/ncs/zephyr/build/zephyr/vif.xml
```
but the way ninja works, then it detects you're in a interactive terminal and decides to overwrite such info, so that you don't see it.

If you redirect to a file then you'll see the line in the file.
Or you can simply do:
```
$ $ ninja -Cbuild |cat
ninja: Entering directory `build'
[1/178] Preparing syscall dependency handling

[2/178] Generating include/generated/version.h
-- Zephyr version: 3.2.99 (/projects/github/ncs/zephyr), build: zephyr-v3.2.0-2030-g9c5b5ca224fc
[3/178] Generating misc/generated/syscalls_subdirs.trigger
[4/178] Generating vif.xml at /projects/github/ncs/zephyr/build/zephyr/vif.xml
....
```

so this has not anything to do with the generator, but how `ninja` itself works.

Please follow existing principle, as the `vif` information is not more important that any other information "hidden" by ninja.

Users who want to know everything that is happening has other options, like use `ninja -v` or `|cat` or simply switch to `make`.
I like the highly reduced output printing with ninja compared to `make`.

Note, please use `Generating` to say consistent with other output messages.

Reply:
also note, other commands doesn't print absolute path, they just print relative path from build folder root.

Reply:
I updated the the comment as suggested in new revision. Thank you